tar/siginfo.c

```diff
@@ -161,11 +162,14 @@ siginfo_printinfo(struct bsdtar *bsdtar, off_t progress)
 				    siginfo->size);
 			}
 		}
+		/* --verbose mode doesn't want newlines at the end of lines. */
 		if (!bsdtar->verbose)
 			fprintf(stderr, "\n");
-		siginfo_received = 0;
 	}
 
+	/* We've handled the signal. */
+	siginfo_received = 0;
```

Comment:
Why did this move?

I don't think we want it here -- if someone hits ^T before we have path and oper set, we don't want to clear `siginfo_received` without responding to it.

Reply:
Hmm.  I was thinking that it would sometimes be useful to print a message if we *don't* have `path` and `oper`, but you're quite correct that it would change the current behaviour so it doesn't belong in this PR.  Removed and I'll check this again to see whether I want to include it in #313.